wp-content/themes/core/assets/pcss/layout/default.pcss

```diff
@@ -0,0 +1,33 @@
+/* -------------------------------------------------------------------------
+ *
+ * Layouts
+ *
+ * ------------------------------------------------------------------------- */
+
+/* -------------------------------------------------------------------------
+ *
+ * Layout: Align Wide
+ *
+ * Adjustments for fluid value in theme.json settings.layout.wideSize
+ *
+ * ------------------------------------------------------------------------- */
+
+/*
+ * Max-width of blocks aligned wide should not account for grid margin
+ * when nested in .has-global-padding because the container applies
+ * grid margin as padding.
+ */
+.has-global-padding.is-layout-constrained > .alignwide,
+:is(.editor-styles-wrapper) .is-root-container.has-global-padding.is-layout-constrained > .alignwide {
+	max-width: 100%;
+}
+
+/*
+ * Max-width of blocks aligned wide nested inside two .has-global-padding
+ * containers need to re-account for grid-margin.
+ * Example: Columns block aligned wide inside group block aligned full.
+ */
+.has-global-padding :is(.has-global-padding) > .alignwide,
+:is(.editor-styles-wrapper) .has-global-padding :is(.has-global-padding) > .alignwide {
+	max-width: var(--wp--style--global--wide-size);
+}
```

Comment:
In this case do the inner blocks even get padding? I couldn't tell from the video but it seemed like the padding had been removed in that second example and that if you responded the browser that it would be full width eventually anyway. This is mostly just me confirming that we would have padding in this situation and that we wouldn't run into responsive issues.